lib/ssl/src/ssl.erl

```diff
@@ -1108,7 +1109,7 @@ append_cipher_suites(Filters, Suites) ->
     (Suites -- Deferred) ++  Deferred.
 
 %%--------------------------------------------------------------------
--spec signature_algs(Description, Version) -> [signature_algs()] when
+-spec signature_algs(Description, Version) -> signature_algs() when
```

Comment:
On a related note: This function may also return `{error, term()}`. It seems a little unfortunate to make the spec `signature_algs() | {error, reason()}`, since that doesn't exactly align with the common style.

Reply:
Humm ... I think the error cases was added later, and I think the correct way for this function will be to change the error cases to erlang:error(Reason) so that you get a better error reason than function clause.  Please add that  to the PR?

Reply:
Yes, that makes sense. I've updated the PR.